pkg/filterengine/filters/node_event_checker.go

```diff
@@ -0,0 +1,73 @@
+// NodeEventsChecker filter to send notifications on critical node events
+
+package filters
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/events"
+	"github.com/infracloudio/botkube/pkg/filterengine"
+	apiV1 "k8s.io/api/core/v1"
+
+	log "github.com/infracloudio/botkube/pkg/logging"
+)
+
+const (
+	// NodeNotReady EventReason when Node is NotReady
+	NodeNotReady string = "NodeNotReady"
+	// NodeReady EventReason when Node is Ready
+	NodeReady string = "NodeReady"
+)
+
+// NodeEventsChecker checks job status and adds message in the events structure
+type NodeEventsChecker struct {
+	Description string
+}
+
+// Register filter
+func init() {
+	filterengine.DefaultFilterEngine.Register(NodeEventsChecker{
+		Description: "Sends notifications on node level critical events.",
+	})
+}
+
+// Run filers and modifies event struct
+func (f NodeEventsChecker) Run(object interface{}, event *events.Event) {
+
+	// Check for Event object
+	eventObj, ok := object.(*apiV1.Event)
+	if !ok {
+		return
+	}
+
+	// Run filter only on Node events
+	if event.Kind != "Node" {
+		// Allow Warning events for all resources
+		if strings.ToLower(eventObj.Type) != config.AllowedEventType.String() {
```

Comment:
I don't think this is the right place to have this check. Let's handle it in the following way:
- Add new event type `InfoEvent EventType = "info"` in config.go. Also change `AllowedEventType EventType = ErrorEvent`
- While calling sendEvent() from Event informer handlers (https://github.com/infracloudio/botkube/blob/develop/pkg/controller/controller.go#L60), set `eventType=config.ErrorEvent` for `Warning` events and `eventType=config.InfoEvent` for `Normal` events
- In event.New(), treat InfoEvent same as ErrorEvent
- In this filter, we should just change event type from InfoEvent -> ErrorEvent in the switch case writen below.
- Finally, in controller, after running all filters,  we should add a check to send only ErrorEvent types.
```
if strings.ToLower(eventObj.Type) != config.AllowedEventType.String() {
    return
}
```
This approach will allow us to create filters for sending particular events as Critical events for other resources also

Reply:
resolved requested changes